extensions/src/test/java/org/pentaho/platform/plugin/services/email/EmailServiceTest.java

```diff
@@ -0,0 +1,479 @@
+    msg.setHeader( "X-Mailer", "smtpsend" );
+    msg.setSentDate( new Date() );
+    try {
+      emailService.sendEmail( emailConfigOriginal, session, msg );
+    } catch ( EmailServiceException e ) {
+      assertTrue( e.getMessage().contains( "401 Unauthorized" ) );
+    }
+  }
+
+  @Test
+  public void testSendEmailSmtp() throws Exception {
+    MockWebServer server = new MockWebServer();
+    String filename = "EmailService/EmailAuthenticationResponse1.json";
+    String content = readTestFile( filename );
+
+    server.enqueue( new MockResponse()
+            .setResponseCode( HttpStatus.SC_OK )
+            .setBody( content )
+    );
+
+    String urlPathToken = "/common/oauth2/v2.0/token";
+    String mockUrlFulToken = server.url( urlPathToken ).toString();
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "test@pentaho.com", "Pentaho Scheduler", "smtp.office365.com", 587,
+                    EmailConstants.PROTOCOL_SMTP, true, "PentahoBA_mailTest@Hitachivantara.com", "", true, true );
+    emailConfigOriginal.setAuthMechanism( EmailConstants.AUTH_TYPE_XOAUTH2 );
+
+    Properties emailProperties = new Properties();
+    emailProperties.setProperty( "mail.smtp.host", emailConfigOriginal.getSmtpHost() );
+    emailProperties.setProperty( "mail.smtp.port", ObjectUtils.toString( emailConfigOriginal.getSmtpPort() ) );
+    emailProperties.setProperty( "mail.transport.protocol", emailConfigOriginal.getSmtpProtocol() );
+    emailProperties.setProperty( "mail.smtp.starttls.enable", ObjectUtils.toString( emailConfigOriginal.isUseStartTls() ) );
+    emailProperties.setProperty( "mail.smtp.ssl", ObjectUtils.toString( emailConfigOriginal.isUseSsl() ) );
+    emailProperties.setProperty( "mail.debug", ObjectUtils.toString( emailConfigOriginal.isDebug() ) );
+    emailConfigOriginal.setClientId( "cid" );
+    emailConfigOriginal.setClientSecret( "secret" );
+    emailConfigOriginal.setTokenUrl( mockUrlFulToken  );
+    emailConfigOriginal.setScope( "https://graph.microsoft.com/.default" );
+    emailConfigOriginal.setGrantType( EmailConstants.GRANT_TYPE_AUTH_CODE );
+    emailConfigOriginal.setAuthorizationCode( "testCode" );
+    emailConfigOriginal.setRedirectUri( "uri" );
+    Session session = Session.getInstance( emailProperties );
+    MimeMessage msg = new MimeMessage( session );
+    msg.setFrom( new InternetAddress( emailConfigOriginal.getDefaultFrom(), emailConfigOriginal.getFromName() ) );
+    msg.setRecipients( Message.RecipientType.TO, InternetAddress.parse( emailConfigOriginal.getDefaultFrom() ) );
+    msg.setSubject( "SUBJECT" );
+    msg.setText( "EmailService.MESSAGE" );
+    msg.setHeader( "X-Mailer", "smtpsend" );
+    msg.setSentDate( new Date() );
+    try {
+      emailService.sendEmail( emailConfigOriginal, session, msg );
+    } catch ( EmailServiceException e ) {
+      assertTrue( e.getMessage().contains( "Authentication unsuccessful" ) );
```

Comment:
for context please document in a comment in this test case `testSendEmailSmtp()`, somewhere why the setup would result in a 401